packages/react-server/src/ReactFlightHooks.js

```diff
@@ -40,7 +40,9 @@ export function prepareToUseHooksForComponent(
 }
 
 export function getThenableStateAfterSuspending() {
-  return thenableState;
+  const state = thenableState;
+  thenableState = null;
+  return state;
```

Comment:
I changed this to reset after a suspend so that if it's the last one it doesn't "leak". Probably not a big deal either way since it gets cleared as soon as something else renders.